### PR TITLE
Update CLM-DART for Derecho

### PR DESCRIPTION
**Pull request by braczka**
## Description:

Primarily shell scripting updates to customize to Derecho.  Also will update CLM and CLM Tutorial documentation.  Will bring in the SourceMod CLM-DART code to version control instead of downloading through image server.

Because of delays when initializing the CAM reanalysis files through campaign/collections directory, reinstituted both 'complete' and 'single_year' datm streamlist files in shell scripts.

### Fixes issue
Address issue #610 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Documentation changes needed?

- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.

### Tests
Performed CLM-DART tutorial and verified results are reproduceable.   

## Checklist for merging

- [ ] Updated changelog entry
- [x] Documentation updated
- [ ] Update conf.py

## Checklist for release
- [ ] Merge into main
- [ ] Create release from the main branch with appropriate tag
- [ ] Delete feature-branch

## Testing Datasets

- [ ] Dataset needed for testing available upon request
- [ ] Dataset download instructions included
- [x] No dataset needed

### CODE REVIEW

**Comment by braczka** [comment]
Draft PR for now.  Need to bring in documentation changes as well as SourceMod code for version control.

**Comment by braczka** [comment]
Documentation changes completed.  Only need to bring in SourceMod code before ready for review.

**Comment by braczka** [comment]
CLM-DART SourceMods have been added for cesm2_2, cesm1_1 and cesm1_2.  The CLM-DART Tutorial works with cesm2_2 SourceMod changes and current research is performed with this version or later.   Older CLM SourceMod changes were brought along for completeness, but I am unaware of any users still using these older versions.

Put in request with cislhelp to further evaluate sluggish response time with /glade/campaign/collections/rda.  Current code partially addresses this by customizing staged reanalysis files based on run time.

**Comment by braczka** [comment]
Ready for review.  Only requires 1 review -- whatever most convenient for Marley and Helen.

**Comment by hkershaw-brown** [comment]
Hi Brett, Great work getting this going on Derecho.

A couple of questions on the SourceMods

1. Are these SourceMods applicable to other models than CLM?  If so is 
   DART/models/CESM/ a good location to put the sourceMods in?

2. The documentation is has instructions for getting the sourcemods from www.image.ucar.edu:
https://dart-documentation--611.org.readthedocs.build/en/611/models/clm/readme.html#sourcemods
https://dart-documentation--611.org.readthedocs.build/en/611/models/clm/tutorial/README.html#adding-clm5-sourcemods
I think this can be changed to 'get the sourcemods from the directory in DART'

**Comment by braczka** [comment]
> Hi Brett, Great work getting this going on Derecho.
> 
> A couple of questions on the SourceMods
> 
> 1. Are these SourceMods applicable to other models than CLM?  If so is
>    DART/models/CESM/ a good location to put the sourceMods in?
> 2. The documentation is has instructions for getting the sourcemods from [www.image.ucar.edu](http://www.image.ucar.edu):
>    https://dart-documentation--611.org.readthedocs.build/en/611/models/clm/readme.html#sourcemods
>    https://dart-documentation--611.org.readthedocs.build/en/611/models/clm/tutorial/README.html#adding-clm5-sourcemods
>    I think this can be changed to 'get the sourcemods from the directory in DART'

These SourceMods are more CLM specific.  Although SourceMods contain directories for other parts of CESM beyond CLM, this is more the accepted practice, and many of these non-CLM directories are empty, especially for cesm2_2_0.

**Review by hkershaw-brown — APPROVED** [review]
Looks good Brett,

I switched out the remain image.ucar.edu reference from clm/readme.rst
see https://github.com/NCAR/DART/pull/611/commits/9ffb633b091c4bcba56c1c390e0a32941a1f116f

 Approved!

File: CHANGELOG.rst [hunk 1]
```diff
@@ -22,6 +22,22 @@ individual files.
 
 The changes are now listed with the most recent at the top.
 
+**January 9 2023 :: Derecho CLM-DART. Tag v10.10.1**
```
Comment (hkershaw-brown) [inline CHANGELOG.rst:25]:
```suggestion
**January 9 2024 :: Derecho CLM-DART. Tag v10.10.1**
```

**Pull request merged**